nerfstudio/configs/method_configs.py

```diff
@@ -96,6 +96,8 @@
     mixed_precision=True,
     pipeline=VanillaPipelineConfig(
         datamanager=VanillaDataManagerConfig(
+            _target=VanillaDataManager[DepthDataset],
```

Comment:
This should be added to `depth-nerfacto` unless we find that it is consistently better and the speed impact is negligible.

Reply:
Regarding speed I find that it doesn't really affect the speed at all, since the loss is only like 4 vector operations.

This is an example of training a scene and the depth rank loss kicks in at 500 training steps. You can see here the average speed doesn't really change, maybe a few more ms per step.

<img width="540" alt="Screen Shot 2023-06-13 at 11 30 35 PM" src="https://github.com/nerfstudio-project/nerfstudio/assets/71362382/f8deb74f-4745-4e70-a7ef-7eb90f287154">

Regarding the quality, I have found from the scenes that we have used it on that it has only helped, but our scenes were primarily tabletop scenes that lack a lot of detail. From the few other scenes I've tried it on, it can smooth out fine features, like the bristles in the brush that I sent above are slightly smoothed out. Definitely requires more testing to get an understanding of how it affects quality.

Reply:
Yea it's better to keep it constrained to depth-nerfacto, it'll wreck performance on things like transparent scenes, fluid/smoke effects, maybe thin features like leaves/foliage, and no clue how it would do on strange objects/scenes.

Reply:
Moved the loss to depth_nerfacto and removed it from nerfacto, added a check to switch between metric depth loss and rank depth loss, if depth is generated or loaded by zoe_depth